tests/orchestrator/orchestrator.ts

```diff
@@ -0,0 +1,88 @@
+import { loadConfig } from "./loadTestConfig";
+import { execSync } from "child_process";
+import fs from "fs";
+import path from "path";
+import { Tenant, Config } from "../test.types";
+
+function findTestFiles(pattern: string, dir: string): string[] {
+  let results: string[] = [];
+
+  const files = fs.readdirSync(dir);
+
+  files.forEach((file) => {
+    const filePath = path.join(dir, file);
+    const stat = fs.statSync(filePath);
+
+    if (stat && stat.isDirectory()) {
+      results = results.concat(findTestFiles(pattern, filePath));
+    } else if (file.endsWith(pattern)) {
+      results.push(filePath);
+    }
+  });
+
+  return results;
+}
+
+function runTestsForTenant(tenant: Tenant) {
+  console.log(`Running tests for tenant: ${tenant.name}`);
+
+  // Set environment variables for the current tenant
+  Object.keys(tenant.env).forEach((key) => {
+    process.env[key] = tenant.env[key as keyof typeof tenant.env];
+  });
+
+  const tenantInstanceName = process.env.NEXT_PUBLIC_AGORA_INSTANCE_NAME;
+
+  // Search the tests directory for test files
+  const testFiles = findTestFiles(
+    `${tenantInstanceName}.test.ts`,
+    path.join(__dirname, "..", "..", "tests")
+  );
+
+  console.log(`Found test files for ${tenant.name}:`, testFiles); // Debugging output
+
+  if (testFiles.length === 0) {
+    console.log(`No test files found for tenant: ${tenant.name}`);
+  }
+
+  try {
+    // Execute the tests we found
+    testFiles.forEach((file) => {
+      console.log(`Running test: ${file}`);
+      execSync(`npx jest ${file} --config jest.config.js`, {
+        stdio: "inherit",
+      });
+    });
+
+    return { tenant: tenant.name, status: "passed" };
+  } catch (error) {
+    return { tenant: tenant.name, status: "failed" };
+  }
+}
+
+// Main loop for test runner
+async function main() {
+  const config: Config = loadConfig() as Config;
```

Comment:
I would have expected a `tenant-config.yaml` to load a single tenant's configuration.  Not a set of tenants.  

The loop should be over a set of tenant-config files, for instance all the files in a directory like this one: https://github.com/voteagora/tenants/tree/main/envs/prod